pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -1395,3 +1410,20 @@ func (c *Cluster) getOSDServicePorts() []v1.ServicePort {
 
 	return ports
 }
+
+func osdStartScript(cmd, args []string, interval string) []string {
+	if interval != "" {
+		parsedInterval, err := time.ParseDuration(interval)
+		// ignore any error and use the default interval of 6 hours.
+		if err == nil {
+			osdRestartInterval = parsedInterval
+		}
+	}
+
+	return []string{
+		"/bin/bash",
+		"-c",
+		"-x",
+		fmt.Sprintf(cephOSDStart, strings.Join(cmd, " "), strings.Join(args, " "), osdRestartInterval.Seconds()),
```

Comment:
Changing the interval to seconds because the golang format of duration does not work in bash script `sleep` command.  

`sleep: invalid time interval '6h0m0s'` 
sleep command is expecting something like `6h` or `6h 0m 0s`

Reply:
Does the `Hours()` helper give us what we need?
```suggestion
		fmt.Sprintf(cephOSDStart, strings.Join(cmd, " "), strings.Join(args, " "), int(osdRestartInterval.Hours())),
```

Reply:
Default 6 hrs
```
+ RESTART_INTERVAL=6.00
+ rc=0
+ '[' 0 -eq 0 ']'
+ touch /tmp/osd-sleep
OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6.00 hours. Restart the pod manually once the flapping issue is fixed
+ echo 'OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6.00 hours. Restart the pod manually once the flapping issue is fixed'
+ sleep 6.00h
```

Configured as 10 minutes in the spec. 
```
debug2023-09-05T14:41:14.231+0000 7fec3f159700  0 osd.0 63 prepare_to_stop starting shutdown
+ RESTART_INTERVAL=0.17
+ rc=0
+ '[' 0 -eq 0 ']'
+ touch /tmp/osd-sleep
+ echo 'OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 0.17 hours. Restart the pod manually once the flapping issue is fixed'
OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 0.17 hours. Restart the pod manually once the flapping issue is fixed
+ sleep 0.17h
```